redash/models.py

```diff
@@ -728,9 +728,9 @@ def to_dict(self):
     def unused(cls, days=7):
         age_threshold = datetime.datetime.now() - datetime.timedelta(days=days)
 
-        unused_results = (db.session.query(QueryResult.id).filter(
-            Query.id == None, QueryResult.retrieved_at < age_threshold)
-            .outerjoin(Query))
+        unused_results = db.session.query(QueryResult.id).filter(
+            QueryResult.retrieved_at < age_threshold,
+            ~QueryResultSet.query.filter(QueryResultSet.result_id == QueryResult.id).exists())
```

Comment:
I think my SQA foo is weak, could you explain what that query does?

Reply:
"where no `QueryResultSet` rows refer to this `QueryResult` row"